### DISCUSSION TITLE
AWS Elasticache Serverless + incompatible maxmemory-policy

### DISCUSSION THREAD

**Discussion by chrislgrigg** [taskforcesh/bullmq#2590] [Q&A]
Hello! In the [BullMQ Elasticache docs](https://docs.bullmq.io/guide/redis-tm-hosting/aws-elasticache), it remarks "Use the standard cache-nodes setup (i.e. not the serverless version, as serverless for the moment uses an incompatible maxmemory-policy)". Can someone please elaborate on this? I can't find any documentation about the max-memory policy with Serverless Elasticache.

**Comment by manast** [comment]
According to the documentation it should be possible to configure: https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/ParameterGroups.Redis.html

**Reply by chrislgrigg** [reply]
Not with Elasticache Serverless. See "Configurability" here: https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/WhatIs.deployment.html#WhatIs.deployment.comparing

**Reply by manast** [reply]
Well, then I guess you answered the question yourself :)

**Reply by chrislgrigg** [reply]
Sort of! I know it can’t be changed but I’m curious about what policy it uses and why it’s incompatible. My understanding of Serverless Elasticache is that it grows as needed so I wouldn’t expect “max memory” to be a thing.

**Reply by manast** [reply]
Because noeviction is the only policy that guarantees that no keys will be ever removed by Redis itself. Since this is serverless and in theory the memory increases automatically maybe it will be a de facto noeviction, but better to ask the AWS support to be sure.

**Reply by Armadillidiid** [reply]
According to the [AWS docs](https://docs.aws.amazon.com/AmazonElastiCache/latest/dg/Scaling-serverless.html#Pre-Scaling), ElastiCache Serverless evicts only keys that have a TTL when memory is full and can't scale up. If no such keys exist, writes will fail with OOM errors.

In that case, would you say it's generally safe to use? If you attach a TTL to key, I think it's normal to expect it can be evicted at any point.

**Comment by usualdesigner** [comment]
Yeah this is a real limitation, not a config thing you're missing. BullMQ needs Redis maxmemory-policy set to noeviction — if Redis is allowed to evict keys under memory pressure it can silently drop job data since your job payloads and state live in Redis. Standard ElastiCache lets you attach a custom parameter group where you set maxmemory-policy to noeviction, but Serverless doesn't expose that knob at all, it's fixed. That's the incompatibility - not something you can enable, you'd need non-serverless ElastiCache with a custom parameter group.
